### DISCUSSION TITLE
HTTP Basic Authentication enabled on service, Dashy can't return green light

### DISCUSSION THREAD

**Discussion by sovajri7** [Lissy93/dashy#1179] [Help and Support]
Hi, I got HTTP Basic Authentication enabled with Nginx Proxy Manager on some services and I don't know if it's possible to make Dashy pass this auth because if not, he can't see if service running or not...
As you can see on image behind, all red services are running but behing the Auth
![image](https://github.com/Lissy93/dashy/assets/94976508/1d884b2a-8468-4d2c-aa15-fbaa4651c648)
![image](https://github.com/Lissy93/dashy/assets/94976508/87ca9232-c731-44a3-a7f8-178d413bb38a)

**Comment by instantdreams** [comment]
I have this issue as well.

It is possible to set your url like this:
```
  - title: Nginx Proxy Manager
    description: Reverse Proxy
    icon: nginxproxymanager.svg
    url: https://[username:password]@[npm.example.com]
```
But this means you will have your credentials in the dashy configuration file.

What might be a good idea is for the config.yml file to support secrets, and they could be substituted:
```
    url: https://[!secret npm-username:!secret npm-password]@[npm.example.com]
```

But what might be better is to find an API that Nginx Proxy Manager exposes and call that to get a healthcheck response. I know the docker container can [run a healthcheck](https://nginxproxymanager.com/advanced-config/#docker-healthcheck).
